### ISSUE TITLE
FixedAdjacencyGraphConvolution layer for human action recognition

### ISSUE THREAD

**Issue by raziyev** [stellargraph/stellargraph#1904]
Hello,
I'm a university student and I'm working on a human action recognition project using GCN and 3D skeleton data.
I have a question from you and I would really appreciate it, if you answer me.
I use this layer (https://stellargraph.readthedocs.io/en/stable/api.html#stellargraph.layer.FixedAdjacencyGraphConvolution) with TimeDistributed keras layer (because I have video frames to process) for my GCN layers. When I define the adjacency matrix as eye matrix, the accuracy is about 2 percent better than when I define it based on the connection between joints. I can not explain it. Would you please help me and tell me why this happens?
Is it wrong layer for this task?

**Comment by huonw** [comment]
Thanks for getting in touch! Unfortunately I cannot say much without more details, can you share the code for how you construct and train your model?  Many graph convolution models include "self-loops" in the adjacency matrix, e.g. `A + I` (sometimes with hyper-parameter weight `λ`, like `A + λ I`), are you doing this?

**Comment by raziyev** [comment]
Thanks for your answer. This is my code. The total accuracy is about 60% without the real adjacency matrix (my mean the connection between joint + the eye matrix (A+I)) and with just the eye matrix (I) it becomes 63%.

model = Sequential()    
model.add(TimeDistributed(FixedAdjacencyGraphConvolution(units = 9, A=A,activation='elu'),input_shape=(n_frames,25,3) ))
model.add(TimeDistributed(FixedAdjacencyGraphConvolution(units = 9,  A=A, activation='elu')))   
model.add(TimeDistributed( FixedAdjacencyGraphConvolution(units = 9,  A=A,activation='elu' )))
model.add(TimeDistributed( FixedAdjacencyGraphConvolution(units = 27, A=A, activation='elu')))  
model.add(TimeDistributed( FixedAdjacencyGraphConvolution(units = 27, A=A, activation='elu'))) 
model.add(TimeDistributed(FixedAdjacencyGraphConvolution(units = 27,  A=A,activation='elu')))  
model.add(TimeDistributed( FixedAdjacencyGraphConvolution(units = 81,A=A, activation='elu')))  
model.add(TimeDistributed( FixedAdjacencyGraphConvolution(units = 81, A=A, activation='elu')))  
model.add(TimeDistributed( FixedAdjacencyGraphConvolution(units = 81,A=A,  activation='elu')))  
model.add(Flatten()) 
model.add(Dense(n_classes, activation="softmax"))

I think something is wrong because of low accuracy and the results of adjacency matrices. I've tried different combinations of GCN unit layers, but the result is the same. I'll really appreciate it, to solve my problem.

**Comment by adocherty** [comment]
Hi,

Firstly, I'll confess to not understanding entirely what your model is doing.  A few comments:

*  You are adding a very deep network with 9 layers - for GCN layers these deep networks can often have poor performance.
*  The shaping of the tensors applying time as well as node dimensions in these problems can be tricky. Carefully look at the dimensions of all tensors to ensure the TimeDistrubuted layer is correctly applied across the time axis, as well as the graph convolution being applied over the node axis.
* It might be wise to start with an existing time series model, in particular the TemporalGCN model here https://stellargraph.readthedocs.io/en/stable/demos/time-series/gcn-lstm-time-series.html

I hope this helps!
